### PR TITLE
feat(backend/shared): add ai chat multi task endpoints bb-340

### PR DESCRIPTION
**Pull request by AndriiLevin**
change change/explain one task for tasks array
change the text of the notification for AI Assistant error

### CODE REVIEW

File: apps/backend/src/modules/ai-assistant/libs/helpers/change-tasks/generate-change-task-suggestions-response.helper.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { ChatMessageAuthor, ChatMessageType, HTTPCode } from "shared";
```
Comment (fshabanov) [inline apps/backend/src/modules/ai-assistant/libs/helpers/change-tasks/generate-change-task-suggestions-response.helper.ts:1]: Don't import directly from shared

File: apps/backend/src/modules/ai-assistant/libs/helpers/helpers.ts [hunk 2]
```diff
@@ -14,3 +6,11 @@ export {
 	generateTaskSuggestionsResponse,
 	runSuggestTaskByCategoryOptions,
 } from "./suggest-task-by-category/run-suggest-task-by-category-option.helper.js";
+export {
+	generateChangeTasksSuggestionsResponse,
+	runChangeTasksByCategoryOptions,
+} from "~/modules/ai-assistant/libs/helpers/change-tasks/run-change-task-by-category-options.helper.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/ai-assistant/libs/helpers/helpers.ts:12]:
```suggestion
} from "./change-tasks/run-change-task-by-category-options.helper.js";
```

File: apps/backend/src/modules/ai-assistant/libs/helpers/helpers.ts [hunk 3]
```diff
@@ -14,3 +6,11 @@ export {
...
+export {
+	generateExplainTasksSuggestionsResponse,
+	runExplainTaskOptions,
+} from "~/modules/ai-assistant/libs/helpers/explain-tasks/run-explain-tasks-option.helper.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/ai-assistant/libs/helpers/helpers.ts:16]:
```suggestion
} from "./explain-tasks/run-explain-tasks-option.helper.js";
```

File: apps/backend/src/modules/ai-assistant/libs/helpers/suggest-task-by-category/generate-suggest-task-response.helper.ts [hunk 4]
```diff
@@ -1,12 +1,15 @@
+import { HTTPCode } from "shared";
 import { type z } from "zod";
 
 import { FIRST_ITEM_INDEX } from "~/libs/constants/constants.js";
+import { OpenAIErrorMessage } from "~/libs/modules/open-ai/libs/enums/open-ai-error-template.enum.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/ai-assistant/libs/helpers/suggest-task-by-category/generate-suggest-task-response.helper.ts:5]:
```suggestion
import { OpenAIErrorMessage } from "~/libs/modules/open-ai/open-ai.js";
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/ai-assistant/libs/helpers/helpers.ts (hunk 2 above)
Comment (AndriiLevin) [reply apps/backend/src/modules/ai-assistant/libs/helpers/helpers.ts:12]: auto rename....
fixed

File: apps/backend/src/modules/ai-assistant/ai-assistant.controller.ts [hunk 5]
```diff
@@ -589,8 +595,10 @@ class AIAssistantController extends BaseController {
 	 *                 type: string
 	 *                 description: Identifier for the thread
 	 *                 example: "thread_5kL0dVY9ADvmNz8U33P7qFX3"
-	 *               payload:
-	 *                 $ref: '#/components/schemas/TaskPayload'
+	 *               payload:  # Исправлен отступ здесь
```
Comment (fshabanov) [inline apps/backend/src/modules/ai-assistant/ai-assistant.controller.ts:598]:
```suggestion
	 *               payload: 
```

**Review by fshabanov — APPROVED** [review]

**Review by WilhelmWesser — APPROVED** [review]

**Comment by KotenkoNata** [comment]
Now to send request explain-tasks and change-tasks- I need to send 

```
{
                    “categoryId”: 8,
                    “categoryName”: “Mental”,
                    “description”: “Spend 10 minutes each day practicing mindfulness meditation to help reduce stress and improve your mental clarity. You can use an app or simply focus on your breathing in a quiet space.“,
                    “dueDate”: “2024-09-29T20:11:15.546Z”,
                    “label”: “Daily Mindfulness Session”
                },
``` 

Maybe better without dueDate as we have for suggest-tasks.

**Comment by AndriiLevin** [comment]
I deleted dueDate so there is a need to re-create Assistant.

**Review by fshabanov — APPROVED** [review]

**Pull request merged**